Document Code Origin coverage via Source Code Integration (#38434)
DataDog/documentation: content/en/tracing/code_origin/_index.md

In Trace Explorer, select a span from an enabled service to see Code Origin details on the Overview tab:
{{< img src="tracing/code_origin/code_origin_details_spotlight.png" alt="Code Origin Details" style="width:100%;">}}

## Getting started

### Prerequisites
- [Datadog APM][6] is configured to capture spans.
- [Source Code Integration][7] is enabled.
- Your service meets the [compatibility requirements](#compatibility-requirements).

> Source Code Integration alone gives you some Code Origin coverage automatically, even if your service doesn't meet the compatibility requirements below. See [Code Origin coverage via Source Code Integration](#code-origin-coverage-via-source-code-integration). Enabling the SDK flag for a compatible language and framework gets you full tracer-instrumented coverage.

Reviewer: ```suggestion
If your Source Code Integration is set up, some Code Origin details can be inferred automatically, even if your service doesn't meet the compatibility requirements below. See [Code Origin coverage via Source Code Integration](#code-origin-coverage-via-source-code-integration). Enabling the SDK flag for a compatible language and framework gets you full tracer-instrumented coverage.
```

{{% /tab %}}

{{% tab "Go" %}}

Go services get Code Origin data through [Source Code Integration](#code-origin-coverage-through-source-code-integration) rather than the SDK flag.

| Requirement | Frameworks |
|---|---|

> | [Source Code Integration](/integrations/guide/source-code-integration/) enabled | custom spans, gRPC servers, Gin, Chi, Echo, Fiber, gorilla/mux, OpenTelemetry, OpenTracing |

Reviewer 2: ```suggestion
| [Source Code Integration](/integrations/guide/source-code-integration/) enabled | Custom spans, gRPC servers, Gin, Chi, Echo, Fiber, gorilla/mux, OpenTelemetry, OpenTracing |
```

### Code Origin section is missing

- Verify Code Origin is [enabled](#enable-code-origin) in your SDK configuration.

> - Confirm that your service meets all [compatibility requirements](#compatibility-requirements) (that is, service language, supported frameworks, and minimum tracer version). If it doesn't, you might still get some Code Origin data automatically. See [Code Origin coverage via Source Code Integration](#code-origin-coverage-via-source-code-integration).

Reviewer: ```suggestion
- Confirm that your service meets all [compatibility requirements](#compatibility-requirements) (that is, service language, supported frameworks, and minimum tracer version). If it doesn't, you might still get some Code Origin data automatically if the source code repository is linked in Datadog. See [Code Origin coverage via Source Code Integration](#code-origin-coverage-via-source-code-integration).
```

### Code Origin section is missing

- Verify Code Origin is [enabled](#enable-code-origin) in your SDK configuration.

> - Confirm that your service meets all [compatibility requirements](#compatibility-requirements) (that is, service language, supported frameworks, and minimum tracer version). If it doesn't, you might still get some Code Origin data automatically if the source code repository is linked in Datadog. See [Code Origin coverage through Source Code Integration](#code-origin-coverage-through-source-code-integration).

Reviewer 2: ```suggestion
- Confirm that your service meets all [compatibility requirements](#compatibility-requirements), including service language, supported frameworks, and minimum tracer version. If it doesn't, you might still get some Code Origin data automatically if the source code repository is linked in Datadog. See [Code Origin coverage through Source Code Integration](#code-origin-coverage-through-source-code-integration).
```

- For transpiled Node.js applications (for example, TypeScript), make sure to generate and publish source maps with the deployed application, run Node.js with the [`--enable-source-maps`][10] flag, and use v5.59.0 or newer of the Node.js tracer. Otherwise, code previews will not work. See the Node.js [Source Code Integration][9] documentation for more details.
- Code Origin is designed to reference user code only, but in some cases, third-party code references may slip through. You can report these cases to [Datadog support][13] and help improve these references.

> ## Code Origin coverage through Source Code Integration

Author: Leaving this heading as-is intentionally. Vale flags it because only the first word is capitalized under strict sentence-case, but "Code Origin" and "Source Code Integration" are treated as proper feature names capitalized throughout this page (e.g. "Using Code Origin", "Enable Code Origin"). Lowercasing them here would be inconsistent with the rest of the doc, so keeping the capitalization for consistency.
